include/odp/api/spec/traffic_mngr.h

```diff
@@ -1464,6 +1464,12 @@ typedef struct {
 	 * have the same single strict priority level and this level must be
 	 * in the range 0..max_priority. */
 	uint8_t priority;
+
+	/** The order_keep when TRUE indicates that when events are enqueued
+	 * into this tm queue while holding an ordered queue synchronization
+	 * context maintain the original event order of the source queue.
+	 */
+	odp_bool_t order_keep;
```

Comment:
This could be reworded a bit since it is packets that are enqueued here, not events. How about this;

/*
 * Maintain original packet order of the source queue when enqueuing
 * packets to this queue while holding ordered or atomic queue
 * synchronization context.
 */

odp_boot_t ordered_enqueue;

Reply:
I would be good if this specified a default value (that is written by ofp_tm_queue_params_init()). I think the default value should be true, since that corresponds to the current behaviour.

The commit message says this about the current implementation: "For Packet IO with TM enabled, there is single mode which is equivalent to Direct mode for enqueuing packets." I do not agree with that since the current linux-gen implementation does ensure the ordering. The API is not explicit about it but the word "enqueue" hints at it. And linux-gen ODP was (is?) a reference implementation of the API.

So, I think this patch is not introducing ordered TM enqueue, but the possibility for unordered enqueue(). The change itself is good and useful.